bgpd/bgp_io.c

```diff
@@ -231,8 +229,10 @@ static int bgp_process_reads(struct thread *thread)
 		 */
 		if (ringbuf_remain(ibw) >= pktsize) {
 			struct stream *pkt = stream_new(pktsize);
-			assert(ringbuf_get(ibw, pktbuf, pktsize) == pktsize);
-			stream_put(pkt, pktbuf, pktsize);
+			/* Copy the data to stream buffer */
+			stream_verify(pkt, pktsize);
```

Comment:
I'm okay with this strategy but instead of this, let's do
```
assert(STREAM_WRITEABLE(pkt) < pktsize))
```

Reply:
will try this change along with the measurement of time it takes between receiving packet in i/o thread and packet to be dequeued in the main thread as suggested.